### PR TITLE
JZD integration

### PR DESCRIPTION
**Pull request by bartDecos**
Added customer ID and session ID support to logs so that they can be sent with specific user data from JZD.

### CODE REVIEW

File: Decos.Diagnostics/ILog.cs [hunk 1]
```diff
@@ -29,6 +31,32 @@ public interface ILog
...
+        void Write<T>(LogLevel logLevel, T data, Guid customerID);
+
+        /// <summary>
+        /// Writes structured data to the log with the specified data set in a LoggerContext object.
+        /// </summary>
+        /// <typeparam name="T">The type of data to write.</typeparam>
+        /// <param name="logLevel">The severity of the data.</param>
+        /// <param name="data">The data to log.</param>
+        /// <param name="senderDetails">Object containing data of the sender.</param>
+        void Write<T>(LogLevel logLevel, T data, LoggerContext senderDetails);
```
Comment (Vivelin) [inline Decos.Diagnostics/ILog.cs]: Changing the interface will break existing implementations. It's better to add these as extension methods, if all you do is call other Write methods on the same instance.

File: Decos.Diagnostics.Trace/TraceSourceLogFactory.cs [hunk 2]
```diff
@@ -84,8 +85,8 @@ private TraceSource CreateSource(SourceName name)
             var switchValue = Options.GetLogLevel(name).ToSourceLevels();
             var traceSource = new TraceSource(name, switchValue);
 
-            var listeners = Options.Listeners
-                .Concat(System.Diagnostics.Trace.Listeners.Cast<TraceListener>());
+            var listeners = Options.Listeners;
+                //.Concat(System.Diagnostics.Trace.Listeners.Cast<TraceListener>());
```
Comment (Vivelin) [inline Decos.Diagnostics.Trace/TraceSourceLogFactory.cs]: Why did you remove this? This will prevent output from showing up in the Output Window in Visual Studio.

**Review by Vivelin — CHANGES_REQUESTED** [review]

File: Decos.Diagnostics/ILog.cs (hunk 1 above)
Comment (bartDecos) [reply Decos.Diagnostics/ILog.cs]: Changes made https://github.com/DecosInformationSolutions/Diagnostics/pull/6/commits/ec92872aaa451db5f31da5d4151be4c57c5e4a94 . 
Though I'm inexperienced with extension methods so I'm not sure if I implemented them the right way?

File: Decos.Diagnostics.Trace/TraceSourceLogFactory.cs (hunk 2 above)
Comment (bartDecos) [reply Decos.Diagnostics.Trace/TraceSourceLogFactory.cs]: I had a problem where the logs were sent twice because every listener existed twice.
That problem is solved however I forgot to undo this change 😅 Sorry about this one.
https://github.com/DecosInformationSolutions/Diagnostics/pull/6/commits/acfbe29825ad6982c091891b8abef92f3b91f8ae

**Review by Vivelin — APPROVED** [review]
